script.js

```diff
@@ -1,4 +1,811 @@
+      }
+      // Push card object to deck array
+      deck.push(card);
+    }
+  }
+  return deck;
+};
+
+// Functions to make a shuffled blackjack deck
+const getRandomNumber = function (max) {
+  return Math.floor(Math.random() * max);
+};
+const shuffleDeck = function (deck) {
+  // Loop through the entire deck
+  for (let currentIndex = 0; currentIndex < deck.length; currentIndex++) {
+    // Get the current card
+    let currentCard = deck[currentIndex];
+    // Get a random index number
+    let randomIndex = getRandomNumber(deck.length);
+    // Get a random card
+    let randomCard = deck[randomIndex];
+    // Swap the random card with the current card
+    deck[currentIndex] = randomCard;
+    deck[randomIndex] = currentCard;
+  }
+  return deck;
+};
+
+// --- The term 'user' is explicitly used to refer to either player or pc ---
+
+// Function to draw multiple random cards
+const drawCards = function (cardCount, userCards, deck) {
+  for (let i = 0; i < cardCount; i++) {
+    // Remove card object from the deck and push it to another array of player or pc's cards
+    userCards.push(deck.pop());
+  }
+  return userCards;
+};
+
+// Function to calculate each user's scores, i.e. sum of all their card values
+const calculateScores = function (userCards) {
+  // Initialize sum to 0
+  let sum = 0;
+  // Loop through the user card values and increment the sum by each of the card values
+  for (let i = 0; i < userCards.length; i++) {
+    sum += userCards[i].rank;
+  }
+
+  // Loop through the user cards to check if there is any ace card and if the sum from the previous loop is > 21, if so, reduce sum by 10 for each ace until the sum is <= 21. Examples are available the bottom of this file.
+  // Put it simply, reduce the sum by 10 only if the ace card causes the user to be busted. Ace is counted as 1 instead of 11 in that case
+  for (let j = 0; j < userCards.length; j++) {
+    if (userCards[j].rank === 11 && sum > 21) {
+      sum -= 10;
+    }
+  }
+  return sum;
+};
+
+// Function to show all the cards each user draws in a one line sentence
+const getAllCardsInfo = function (userCards) {
+  // Initialize a string to hold each user's cards info
+  let userCardsInfo = "";
+  for (i = 0; i < userCards.length; i++) {
+    // Update the string by each card name and suit
+    userCardsInfo += `<em>${userCards[i].name}</em> of ${userCards[i].suit},   `;
+  }
+  return userCardsInfo;
+};
+
+// Function to analyze winning conditions and generate result output message
+// First check the strictest condition, i.e. if there is any immediate win by blackjack , then check if anyone has busted, then check the remaining normal conditions
+const getResult = function (
+  playerCards,
+  pcCards,
+  playerScore,
+  pcScore,
+  WINNING_SCORE
+) {
+  // Initialize a string to store result info which will be combined with the standard output message later
+  let result = `<hr> Please drag the slider to bet and press Play to continue playing.`;
```

Comment:
\nit somehow find that `result` doesn't really convey what this variable stores, cause there's other text/instructions that are not exactly the result, maybe `outputMessage` or `message`?